controllers/front/callbacknotification.php

```diff
@@ -22,124 +22,151 @@ public function postProcess()
-                    if ($response && is_array($response->Transactions)) {
-                        $order_status = (int) Configuration::get('PS_OS_PAYMENT');
-                        $currency_paid = Currency::getIdByIsoCode($response->Currency);
-                        $amount_paid = $response->amount;
-                        $paymentType = $response->Transactions[0]->AuthType;
-                        /* If payment type is 'payment' funds have not yet been captured,
-                        * so AltaPay returns 0 as the captured amount.Therefore we assume full payment has been authorized.
-                        */
-                        if ($paymentType === 'payment') {
-                            $amount_paid = $cart->getOrderTotal(true, Cart::BOTH);
-                            $currency_paid = new Currency($cart->id_currency);
-                        }
-                        // Determine payment method for display
-                        $paymentMethod = determinePaymentMethodForDisplay($response);
 
-                        // Create an order with 'payment accepted' status
-                        $cSk = $customer->secure_key;
-                        $cpId = (int) $currency_paid->id;
-                        $cId = $cart->id;
-                        $oSt = $order_status;
-                        $pMeth = $paymentMethod;
-                        $this->module->validateOrder($cId, $oSt, $amount_paid, $pMeth, null, null, $cpId, false, $cSk);
-                        // Log order
-                        $current_order = new Order((int) $this->module->currentOrder);
+            //Set order status, if available from the payment gateway
+            $cardHolderMessageMustBeShown = false;
+            if (isset($postData['cardholder_message_must_be_shown'])) {
+                $cardHolderMessageMustBeShown = $postData['cardholder_message_must_be_shown'];
+            }
+            if (isset($postData['error_message']) && $cardHolderMessageMustBeShown == 'true') {
+                $msg = $postData['error_message'];
+            } else {
+                $msg = 'Error with the Payment.';
+            }
+
+            if ($resultStatus == 'cancelled') {
+                $msg = 'Payment canceled';
+            }
+
+            switch ($resultStatus) {
+                case 'succeeded':
+                case 'success':
+                    $this->handleNotificationAction($cart, $order, $response, $customer, $transactionStatus, $shopOrderId);
+                    break;
+                case 'cancelled':
+                    $this->handleCancelledStatusAction($shopOrderId, $transactionStatus);
+                    break;
+                case 'error':
+                case 'failed':
+                    $this->handleFailedStatusAction($msg, $paymentId, $order, $transactionStatus);
+                    break;
+                default:
+                    $this->handleUnExpectedStatusAction($shopOrderId, $transactionStatus);
```

Comment:
What about the Open status? https://documentation.altapay.com/Content/Ecom/Payment%20Pages/Payment%20Page%20Open.htm?Highlight=open

Reply:
The open status means that the order has been created, but has not been settled yet.

Reply:
when the order status is open it will not trigger the callbacknotification.php file instead it will trigger the callbackopen.php file, which is already handling this status. Here is the link https://github.com/AltaPay/plugin-prestashop/blob/master/controllers/front/callbackopen.php